File: Tests/CSF.Screenplay.Web.Tests/Pages/PageTwo.cs
using System;
using CSF.Screenplay.Web.Models;

namespace CSF.Screenplay.Web.Tests.Pages
{
  public class PageTwo : Page
  {
    public override string GetName() => "the second page";

    public override IUriProvider GetUriProvider() => new AppUri("PageTwo");

    public static ITarget SpecialInputField => new CssSelector(".special_text input", "the special input field");

    public static ITarget SecondTextbox => new CssSelector(".second_textbox input", "the second text box");

    public static ITarget TheDynamicTextArea => new ElementId("dynamic_value", "the dynamic value");

    public static ITarget SingleSelectionList => new CssSelector("#single_selection", "the single selection list");

    public static ITarget SingleSelectionValue => new CssSelector("#single_selected_value", "the single-selection value");

    public static ITarget MultiSelectionList => new CssSelector("#multiple_selection", "the multi selection list");

    public static ITarget MultiSelectionValue => new CssSelector("#multiple_selected_value", "the multi-selection value");

    public static ITarget ListOfItems => new CssSelector("#list_of_items", "the list of items");

    public static ILocatorBasedTarget ItemsInTheList => new CssSelector("#list_of_items li", "items in the list");

    public static ITarget JavaScriptResult => new ElementId("ScriptOutput", "the Javascript output");
  }
}

File: CSF.Screenplay.WebTestWebsite_old/Scripts/PageTwo.js
(function(window, $, undefined) {
  'use strict';

  var
    specialInputField,
    scriptOutput,
    singleSelection,
    singleSelectionValue,
    multiSelection,
    multiSelectionValue;

  function setupElements()
  {
    specialInputField = $('.special_text input');
    scriptOutput = $('#ScriptOutput');
    
    specialInputField.on('change keyup', function() {
      if(specialInputField.val() === 'The right value')
      {
        $('#dynamic_value').text('different value');
      }
      else
      {
        $('#dynamic_value').text('dynamic value');
      }
    });
    
    singleSelection = $('#single_selection');
    singleSelectionValue = $('#single_selected_value');
    multiSelection = $('#multiple_selection');
    multiSelectionValue = $('#multiple_selected_value');

    singleSelection.on('change click', function() {
      recalculateSelections(singleSelection, singleSelectionValue);
    });
      
    multiSelection.on('change click', function() {
      recalculateSelections(multiSelection, multiSelectionValue);
    });
  }

  function recalculateSelections(selection, value)
  {
    var val = selection.val();
    
    if(!val || !val.length)
      value.text('Nothing!');
    else
      value.text(val);
  }

  function addCallableScripts()
  {
    window.myCallableScript = function()
    {
      scriptOutput.text('myCallableScript called');
    }
    
    window.addFive = function(inputVal)
    {
      if(!inputVal)
      {
        inputVal = 1;
        scriptOutput.text('addFive called without any input');
      }
      else
      {
        scriptOutput.text('addFive called with ' + inputVal);
      }
      
      return inputVal + 5;
    }
  }
  
  function init()
  {
    setupElements();

    recalculateSelections(singleSelection, singleSelectionValue);
    recalculateSelections(multiSelection, multiSelectionValue);
    addCallableScripts();
  }
  
  $(function() {
    init();
  });
  
}(window, jQuery));